modules/create_timepoint/php/ajaxtimepoint.class.inc

```diff
@@ -0,0 +1,309 @@
+            }
+            if (empty($subprojList)) {
+                $errors['subprojectID'] = 'No subprojects have been 
+                defined for the project this candidate is affiliated with. 
+                If you are an administrator, please use the Configuration module to 
+                add new subprojects and associate them with projects.';
+            }
+        }
+
+        // Retrieve visit labels.
+        $visit_options      = array();
+        $visitLabelSettings = $config->getSetting('visitLabel');
+        foreach (
+            \Utility::toArray($visitLabelSettings) as $visitLabel
+        ) {
+            if (!empty($values['subproject'])) {
+                if (isset($visitLabel['generation'])
+                    && $visitLabel['generation'] !== 'sequence'
+                ) {
+                    throw new \DeprecatedException(
+                        'User generated Visit Labels are no '.
+                        'longer supported in Loris.'
+                    );
+                }
+                $visitLabel       = \Utility::toArray($visitLabel)[0];
+                $labelOptions[''] = null;
+                $items            = \Utility::toArray(
+                    $visitLabel['labelSet']['item']
+                );
+                foreach ($items as $item) {
+                    $item = \Utility::toArray($item)[0];
+                    $labelOptions[$item['@']['value']] = $item['#'];
+                }
+                $visit_options[$visitLabel['@']['subprojectID']]
+                    = array_filter($labelOptions);
+            }
+        }
+        $values['visit'] = $visit_options;
+        if (!empty($visit_options)) {
+            // List of sites for the user.
+            $user = \User::singleton();
+            $DB   = \Database::singleton();
+            $user_list_of_sites = $user->getData('CenterIDs');
+            $num_sites          = count($user->getData('CenterIDs'));
+            $psc_labelOptions   = array();
+            if ($num_sites > 1) {
+                $psc_labelOptions = array(null => '');
+                foreach ($user_list_of_sites as $key => $siteID) {
+                    $center = $DB->pselectRow(
+                        "SELECT CenterID as ID, Name FROM psc 
+                        WHERE CenterID =:cid",
+                        array('cid' => $siteID)
+                    );
+                    $psc_labelOptions[$siteID] = $center['Name'];
+                }
+            }
+            $values['psc'] = array_filter($psc_labelOptions);
+        }
+        if (!empty($errors)) {
+            $values['errors'] = $errors;
+        }
+        if (empty($values['psc'])) {
+            unset($values['psc']);
+        }
+
+        return $values;
+    }
+
+    /**
+     * Validate the post data,
+     * from user user form selection.
+     *
+     * @param array $values the form values.
+     *
+     * @return array
+     * @throws \LorisException
```

Comment:
This doesn't throw a LorisException

Reply:
@johnsaigle `\TimePoint::isValidVisitLabel($candid, $subprojectID, $visitLabel);` can possibly throw the error.

Reply:
But you catch that function call so it can't actually throw this exception

Reply:
@johnsaigle yes but I believe the point of the @throws are for making the process of debugging a problem simpler. Since you can navigate from method to method and find where a problem is because an Exception is usually what breaks functionality. Specifically if we're logging things correctly and @throws help in navigating the code as well.

Reply:
Yeah but it doesn't actually throw the Exception lol. It feels very misleading to me.

I understand it can help you navigate in your IDE but I just use Vim and several others do also so your example doesn't really provide that much utility to several of us. 

I think this is part of the discussion to be had in the LORIS meeting that we mentioned above, don't worry about this until a decision is made there.

Reply:
@johnsaigle I understand and I don't necessarily use the @throw tags myself and when debugging where an exception is thrown. Typically I use breakpoints to see what is causing the code to break. The only point I would voice at the meeting in favour of @throw tags in functions that use functions that possibly throw and exception is how I think it's best practice because it's helpful in the sense of knowing wherever the starting point is where an possible exception can be thrown. To me it's similar to the php functions type hinting in the sense that it's optional but useful in a small way.

Reply:
@johnsaigle also I think the comments in general for functions in LORIS need to be cleaned up. I've noticed some core functions with comments including @throw exceptions and it's no longer the case anymore. I'll make a PR in the future to clean some of it up.